rust/src/lib/unit_tests/gen_diff_test_files/change_dns_only/diff.yml

```diff
@@ -0,0 +1,11 @@
+---
+dns-resolver:
+  config:
+    search:
+    - example.com
```

Comment:
I can not see the dns-search difference between the current state and the desired state with the example provided except that the DNS server order changed.

Reply:
Aha, the `gen_diff()` just copies all the DNS settings over when the DNS settings are changed.

```
        if merged_state.dns.is_changed() {
            ret.dns = self.dns.clone();
        }
```

Reply:
This seems harmless. Let's fix that in follow up PR. Issue tracked at https://github.com/nmstate/nmstate/issues/2559